File: Easy/Array/1365. How Many Numbers Are Smaller Than the Current Number/Solution.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] count = new int[101];

        for(int num : nums){
            count[num]++;
        }

        //prefix sum
        for(int i = 1; i < 101; i++){
            count[i] += count[i - 1];
        }

        int[] res = new int[nums.length];

        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                res[i] = 0;
            } else {
                res[i] = count[nums[i] - 1];
            }
        }

        return res;
    }
}

File: Easy/Array/1365. How Many Numbers Are Smaller Than the Current Number/Solution.kt
class Solution {
    fun smallerNumbersThanCurrent(nums: IntArray): IntArray {
        
        val freq = IntArray(101)

        //count occurences
        for(num in nums) {
            freq[num]++
        }

        // convert freq into how many values are smaller
        for(i in 1..100) {
            freq[i] += freq[i - 1]
        }

        val result = IntArray(nums.size)

        for(i in nums.indices){
            result[i] = if(nums[i] == 0) {
                0
            } else {
                freq[nums[i] - 1]
            }
        }

        return result
    }
}
